### PR TITLE
Add gateway_id to the config flow title.

### PR DESCRIPTION
**Pull request by vlebourl**

### CODE REVIEW

File: custom_components/tahoma/config_flow.py [hunk 1]
```diff
@@ -124,36 +124,35 @@ async def async_step_reauth(
...
 
-    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo):
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
         """Handle DHCP discovery."""
         hostname = discovery_info.hostname
         gateway_id = hostname[8:22]
 
         _LOGGER.debug("DHCP discovery detected gateway %s", obfuscate_id(gateway_id))
-
-        if self._gateway_already_configured(gateway_id):
```
Comment (iMicknl) [inline custom_components/tahoma/config_flow.py:134]: Could you remove this function as well? `_gateway_already_configured`

File: tests/test_config_flow.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
+"""Tests for Overkiz (by Somfy) config flow."""
 from __future__ import annotations
 
 from unittest.mock import Mock, patch
 
 from aiohttp import ClientError
-from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant import config_entries, data_entry_flow
 from homeassistant.components import dhcp
+from homeassistant.components.overkiz.const import DOMAIN
```
Comment (iMicknl) [inline tests/test_config_flow.py:9]: Please change the tests for our custom integration ;-).

File: tests/test_config_flow.py [hunk 3]
```diff
@@ -57,7 +60,7 @@ async def test_form(hass):
     with patch("pyoverkiz.client.OverkizClient.login", return_value=True), patch(
         "pyoverkiz.client.OverkizClient.get_gateways", return_value=None
     ), patch(
-        "custom_components.tahoma.async_setup_entry", return_value=True
+        "homeassistant.components.overkiz.async_setup_entry", return_value=True
```
Comment (iMicknl) [inline tests/test_config_flow.py:63]: Here as well (and all other ones)

File: custom_components/tahoma/config_flow.py [hunk 4]
```diff
@@ -112,56 +124,54 @@ async def async_step_user(
...
-            _LOGGER.debug("Gateway %s is already configured", obfuscate_id(gateway_id))
-            return self.async_abort(reason="already_configured")
-
-        return await self.async_step_user()
+        _LOGGER.debug(
+            "DHCP discovery detected gateway %s on host %s",
+            obfuscate_id(gateway_id),
+            discovery_info.hostname,
```
Comment (iMicknl) [inline custom_components/tahoma/config_flow.py:135]: Why this change? (and do we need to make this in core as well?)

Comment (iMicknl) [reply custom_components/tahoma/config_flow.py:135]: But doesn't this print gateway-xxxxx? (so the raw information we use for gateway id?)

File: custom_components/tahoma/config_flow.py [hunk 5]
```diff
@@ -112,56 +124,54 @@ async def async_step_user(
...
         self, discovery_info: zeroconf.ZeroconfServiceInfo
     ) -> FlowResult:
         """Handle ZeroConf discovery."""
 
         # abort if we already have exactly this bridge id/host
-        # reload the integration if the host got updated
         properties = discovery_info.properties
         gateway_id = properties["gateway_pin"]
 
         _LOGGER.debug(
```
Comment (vlebourl) [inline custom_components/tahoma/config_flow.py:148]: Goes with this one as well.

File: custom_components/tahoma/config_flow.py (hunk 4 above)
Comment (vlebourl) [reply custom_components/tahoma/config_flow.py:135]: hmm possible! I was trying to print the ip address.

Comment (iMicknl) [reply custom_components/tahoma/config_flow.py:135]: Hostname is not the ip address in this case. I think best would to remove both of these and just keep it the same as in core.

Comment (vlebourl) [reply custom_components/tahoma/config_flow.py:135]: goes with 
```py
        _LOGGER.debug(
            "ZeroConf discovery detected gateway %s on host %s",
            obfuscate_id(gateway_id),
            discovery_info.host,
        )
```

**Review by iMicknl — APPROVED** [review]
